tools/converter.py

```diff
@@ -413,14 +412,54 @@ def string_formats(self, argstr, name):
         return argstr_new
 
 
-
-@pytest.mark.parametrize("interface_name",
-                         ["BET", "MCFLIRT", "FLIRT", "FNIRT", "ApplyWarp", "SliceTimer",
-                          "SUSAN", "PRELUDE", "FIRST", "FAST"]
-                         )
-def test_convert_file(interface_name):
-    converter = FSLConverter(interface_name=interface_name)
-
-    dirname_interf = Path(__file__).parent.parent / "pydra/tasks/fsl/preprocess"
-
-    input_spec, output_spec = converter.pydra_specs(write=True, dirname=dirname_interf)
+FSL_MODULES = ['aroma', 'dti', 'epi', 'fix', 'maths', 'model', 'possum', 'preprocess', 'utils']
+
+@click.command()
+@click.option("-i", "--interface_name", required=True, default="all",
+              help="name of the interface (name used in Nipype, e.g. BET) or all (default)"
+                   "if all is used all interfaces from the spec file will be created")
+@click.option("-m", "--module_name", required=True, help=f"name of the module from the list {FSL_MODULES}")
+def create_pydra_spec(interface_name, module_name):
+    if module_name not in FSL_MODULES:
+        raise Exception(f"module name {module_name} not available;"
+                        f"should be from the list {FSL_MODULES}")
+
+    spec_file = Path(os.path.dirname(__file__)) / f"../specs/fsl_{module_name}_param.yml"
+    if not spec_file.exists():
+        raise Exception(f"the specification file doesn't exist for the module {module_name},"
+                        f"create the specification file in {spec_file.parent}")
+
+    def all_interfaces(module):
```

Comment:
```suggestion
    @functools.lru_cache()
    def all_interfaces(module):
```

You'll need to import `functools` up top, but this will keep you from having to re-read the YAML file every time you call `all_interfaces()`.